src/features/usersActionBar/Rating.tsx

```diff
@@ -8,14 +9,98 @@ import {
   loadFilteredUsers,
   selectSearchRatingText,
 } from '../users/usersSlice';
-import { Params } from '../../types/types';
+import { RatingProp } from '../../types/types';
+import Rate from './Rate';
 
-const Rating: FC<Params> = ({ id }: Params) => {
+const Stars = styled.div`
+  cursor: pointer;
+  display: flex;
+  margin: 0 auto;
+
+  &:hover {
+    svg polygon {
+      fill: #ffd055 !important;
+    }
+  }
+
+  &[data-stars] {
+    svg polygon {
+      fill: #ffd055;
+    }
+  }
+
+  &[data-stars="-4"] {
+    svg:nth-child(1) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  &[data-stars="-3"] {
+    svg:nth-child(2) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  &[data-stars="-2"] {
+    svg:nth-child(3) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  &[data-stars="-1"] {
+    svg:nth-child(4) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  &[data-stars="0"] {
+    svg:nth-child(5) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  &[data-stars="1"] {
+    svg:nth-child(6) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  &[data-stars="2"] {
+    svg:nth-child(7) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  &[data-stars="3"] {
+    svg:nth-child(8) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  &[data-stars="4"] {
+    svg:nth-child(9) ~ svg polygon {
+      fill: #d8d8d8;
+    }
+  }
+
+  svg:hover ~ svg {
+    polygon {
+      fill: #d8d8d8 !important;
+    }
+  }
```

Comment:
Ты, конечно, не самый простой подход выбрала для реализации рейтинга :) Но, как минимум, можно хотя бы стили  немного упростить и компонент `Rate` (выше написал).

```suggestion
  &[data-stars] svg {
    color: #ffd055;
  }

  &:hover svg {
    color: #ffd055 !important;
  }

  &[data-stars="-4"] svg:nth-child(1) ~ svg,
  &[data-stars="-3"] svg:nth-child(2) ~ svg,
  &[data-stars="-2"] svg:nth-child(3) ~ svg,
  &[data-stars="-1"] svg:nth-child(4) ~ svg,
  &[data-stars="0"] svg:nth-child(5) ~ svg,
  &[data-stars="1"] svg:nth-child(6) ~ svg,
  &[data-stars="2"] svg:nth-child(7) ~ svg,
  &[data-stars="3"] svg:nth-child(8) ~ svg,
  &[data-stars="4"] svg:nth-child(9) ~ svg {
    color: #d8d8d8;
  }

  svg:hover ~ svg {
    color: #d8d8d8 !important;
  }
```